lib/plug/cowboy/handler.ex

```diff
@@ -15,12 +18,26 @@ defmodule Plug.Cowboy.Handler do
       {:ok, req, {plug, opts}}
     catch
       kind, reason ->
-        exit_on_error(kind, reason, System.stacktrace(), {plug, :call, [conn, opts]})
+        stacktrace = System.stacktrace()
+        metadata = %{kind: kind, error: reason, stacktrace: stacktrace, conn: conn}
+
+        :telemetry.execute(
+          [:plug_cowboy, :handler, :failure],
+          %{duration: System.monotonic_time() - start},
+          metadata
+        )
+
+        exit_on_error(kind, reason, stacktrace, {plug, :call, [conn, opts]})
     after
       receive do
         @already_sent -> :ok
       after
-        0 -> :ok
+        0 ->
+          :telemetry.execute([:plug_cowboy, :handler, :stop], %{
```

Comment:
Nitpick: maybe compute the duration on its own line so we can keep everything in a single line?